src/screens/view-wallet-mnemonic/view-wallet-mnemonic.tsx

```diff
@@ -49,14 +48,15 @@ export class ViewWalletMnemonicScreenComponent extends React.Component<
     }
 
     public componentDidMount() {
-        this.passwordModal
-            .requestPassword()
-            .then(password => {
-                this.populateMnemonic(password);
-            })
-            .catch(() => {
-                this.props.navigation.goBack(null);
-            });
+        PasswordModal.getPassword(
+            translate('Password.pinTitleUnlock'),
+            translate('Password.subtitleMnemonic')
+        ).then(password => this.populateMnemonic(password));
+
```

Comment:
if it fails? This screen should decide if we need to ask the password or not, not the previous